pyomo/contrib/doe/doe.py

```diff
@@ -25,1255 +25,1169 @@
+    # Use kaug to get FIM
+    def _kaug_FIM(self, model=None):
+        """
+        Used to compute the FIM using kaug, a sensitivity-based
+        approach that directly computes the FIM.
 
-        # discretize if needed
-        if self.discretize_model is not None:
-            mod = self.discretize_model(mod, block=False)
+        Parameters
+        ----------
+        model: model to compute FIM, default: None, (self.compute_FIM_model)
 
-        # Deactivate any existing objective functions
-        for obj in mod.component_objects(pyo.Objective):
-            obj.deactivate()
+        """
+        # Remake compute_FIM_model if model is None.
+        # compute_FIM_model needs to be the right version for function to work.
+        if model is None:
+            self.compute_FIM_model = self.experiment.get_labeled_model(
+                **self.get_labeled_model_args
+            ).clone()
+            model = self.compute_FIM_model
 
         # add zero (dummy/placeholder) objective function
-        mod.Obj = pyo.Objective(expr=0, sense=pyo.minimize)
-
-        # set ub and lb to parameters
-        for par in self.param.keys():
-            cuid = pyo.ComponentUID(par)
-            var = cuid.find_component_on(mod)
-            var.setlb(self.param[par])
-            var.setub(self.param[par])
-
-        # generate parameter name list and value dictionary with index
-        var_name = list(self.param.keys())
+        if not hasattr(model, "objective"):
+            model.objective = pyo.Objective(expr=0, sense=pyo.minimize)
 
         # call k_aug get_dsdp function
-        square_result = self._solve_doe(mod, fix=True)
+        # Solve the square problem
+        # Deactivate object and fix experimental design decisions to make square
+        for comp in model.experiment_inputs:
+            comp.fix()
 
-        # save model from optional post processing function
-        self._square_model_from_compute_FIM = mod
+        self.solver.solve(model, tee=self.tee)
 
-        dsdp_re, col = get_dsdp(
-            mod, list(self.param.keys()), self.param, tee=self.tee_opt
-        )
+        # Probe the solved model for dsdp results (sensitivities s.t. parameters)
+        params_dict = {k.name: v for k, v in model.unknown_parameters.items()}
+        params_names = list(params_dict.keys())
+
+        dsdp_re, col = get_dsdp(model, params_names, params_dict, tee=self.tee)
 
         # analyze result
         dsdp_array = dsdp_re.toarray().T
-        self.dsdp = dsdp_array
-        self.dsdp = col
+
         # store dsdp returned
         dsdp_extract = []
         # get right lines from results
         measurement_index = []
 
         # loop over measurement variables and their time points
-        for mname in self.measure_name:
+        for k, v in model.experiment_outputs.items():
+            name = k.name
             try:
-                kaug_no = col.index(mname)
+                kaug_no = col.index(name)
```

Comment:
The `get_dsdp` method from pyomo.contrib.sensitivity_toolbox.sens uses string names. So we would have to change that upstream code to fix this as well. I can add a TODO if we have time later (and if we do more analysis with things like kaug) to update this code to avoid strings, but I do not know who else or what else depends on the get_dsdp method.